dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -310,105 +337,122 @@ public static StoreExclusive newExclusiveStore(Register register, Expression add
     // ========================================== Pthread ==========================================
     // =============================================================================================
 
-    public static class Pthread {
-        private Pthread() {
-        }
+    public Pthread withPthread() {
+        return new Pthread(this);
+    }
+
+    public static final class Pthread extends EventFactory {
```

Comment:
These decorators feel kinda ugly.
What about the classes like `Pthread` not inheriting from `EventFactory` and renaming `withPthread` to `forPthread`.
Then the user will always be maintaining a pure `EventFactory` instance and use `eventFactory.forPthread().newPthreadEvent`.
At least this stays more true to our current usage pattern.
It also disables strange usages like `eventFactory.withLLVM().withPthread().withX86()` which has no reasonable meaning.

Alternatively, we could forgo the whole partioning/grouping of events and have them all in a single `EventFactory`.
This might result in more verbose naming though.

Reply:
Although there currently are no overrides, I would propose this inheritance.  From the perspective of a user of `EventFactory`, I would rather like to have just one or two places where it really matters, if I am creating target-specific events, or core events.  E.g. there is this code in `PowerVisitor.java`:
```
  Load load = eventFactory.newRMWLoadExclusive(oldValueRegister, address);
  Store store = eventFactory.newRMWStoreConditional(address, e.getStoreValue(), true);
```
With distinct classes, this code would currently look like
```
  Load load = eventFactory.newRMWLoadExclusive(oldValueRegister, address);
  Store store = eventFactory.forPower().newRMWStoreConditional(address, e.getStoreValue(), true);
```

Reply:
I understand that with distinct classes it looks different, in particular, more explicit. I'm just advocating for the more explicit version. At the end of the day, this is just my opinion. Functionalitywise, I don't see much difference in both approaches.